src/view.c

```diff
@@ -393,7 +393,7 @@ push_view_history_state(struct view_history *history, struct position *position,
 
 	if (state && data && history->state_alloc &&
 	    !memcmp(state->data, data, history->state_alloc))
-		return NULL;
+		return state;
```

Comment:
Good catch. Return the current stack when nothing changed?

Reply:
Yes, otherwise reopening the same blame fails.